internal/commands/server/command.go

```diff
@@ -172,11 +172,15 @@ func (c *Command) Run(args []string) int {
 		MirrorKubernetesNamespacePrefix: c.flagMirrorK8SNamespacePrefix,
 	}
 
-	consulHTTPAddressOrCommand, port, err := parseConsulHTTPAddress()
+	consulScheme, consulHTTPAddressOrCommand, port, err := parseConsulHTTPAddress()
 	if err != nil {
 		logger.Error("error reading "+consulHTTPAddressEnvName, "error", err)
 		return 1
 	}
+	if consulCfg.Scheme != "https" {
+		// override only if it needs to be explicitly marked
+		consulCfg.Scheme = consulScheme
+	}
```

Comment:
sorry, this is to say, if we already have the scheme set to `https` (via things like ENV variables, which `DefaultConfig` will source), then don't override it -- otherwise  set it based off of what we got from parsing it out of the `CONSUL_ADDR` (in 1.0.0 we don't set the scheme and use `CONSUL_USE_SSL`, in <= 0.49 we use the scheme as part of the address, this handles both cases)